test/basics_datamunger_test.js

```diff
@@ -391,34 +441,234 @@ describe('basics datamunger', function() {
+        },
+        profile: {
+          fullName: 'Jill Jellyfish',
+          patient: {
+              settings: {
+                siteChangeSource: constants.SITE_CHANGE_CANNULA,
+              },
+          },
+        },
+      };
+
+      dm.processInfusionSiteHistory(basicsData, constants.TANDEM, patient);
+      expect(basicsData.sections.siteChanges.type).to.equal(constants.SITE_CHANGE_CANNULA);
+    });
+
+    it('should set siteChanges type to tubingPrime', function() {
+      var basicsData = {
+        data: {
+          [constants.SITE_CHANGE_CANNULA]: {dataByDate: countSiteChangesByDay},
+          [constants.SITE_CHANGE_TUBING]: {dataByDate: countSiteChangesByDay},
+        },
+        days: oneWeekDates,
+        sections: siteChangeSections,
+      };
+
+      var patient = {
+        permissions: {
+          'root': {},
+        },
+        profile: {
+          fullName: 'Jill Jellyfish',
+          patient: {
+            settings: {
+              siteChangeSource: constants.SITE_CHANGE_TUBING,
+            },
+          },
+        },
+      };
+
+      dm.processInfusionSiteHistory(basicsData, constants.TANDEM, patient);
+      expect(basicsData.sections.siteChanges.type).to.equal(constants.SITE_CHANGE_TUBING);
+    });
+
+    it('should set siteChanges type to reservoirChange', function() {
+      var basicsData = {
+        data: {
+          [constants.SITE_CHANGE_RESERVOIR]: {dataByDate: countSiteChangesByDay}
+        },
+        days: oneWeekDates,
+        sections: siteChangeSections,
+      };
+
+      var patient = {
+        permissions: {
+          'root': {},
+        },
+        profile: {
+          fullName: 'Jill Jellyfish',
+          patient: {
+            settings: {
+              siteChangeSource: constants.SITE_CHANGE_TUBING,
+            },
+          },
+        },
+      };
+
+      dm.processInfusionSiteHistory(basicsData, constants.INSULET, patient);
+      expect(basicsData.sections.siteChanges.type).to.equal(constants.SITE_CHANGE_RESERVOIR);
+    });
+
+    it('should set siteChanges type to undeclared, and settings to be open', function() {
+      var basicsData = {
+        data: {
+          [constants.SITE_CHANGE_CANNULA]: {dataByDate: countSiteChangesByDay},
+          [constants.SITE_CHANGE_TUBING]: {dataByDate: countSiteChangesByDay},
+        },
+        days: oneWeekDates,
+        sections: siteChangeSections,
+      };
+
+      var patient = {
+        permissions: {
+          'root': {},
+        },
+        profile: {
+          fullName: 'Jill Jellyfish',
+          patient: {
+            profile: {
+              settings: {},
+            },
+          }
+        },
+      };
+
+      dm.processInfusionSiteHistory(basicsData, constants.TANDEM, patient);
+      expect(basicsData.sections.siteChanges.type).to.equal(constants.SECTION_TYPE_UNDECLARED);
+      expect(basicsData.sections.siteChanges.settingsTogglable).to.equal(togglableState.open);
+    });
+  });
```

Comment:
I'd be interested in a more exhaustive set of conditions here, like all combos of no `siteChangeSource` in settings and pump manufacturer, and also the semi-pathological case of `siteChangeSource` preference not matching the available data, as will happen, potentially, when someone switches pumps.